src/git/run.rs

```diff
@@ -173,6 +173,15 @@ impl GitRunInfo {
 
         let args = {
             let mut result = vec!["-C", repo_path];
+            if let Some(working_copy_path) = repo.get_working_copy_path() {
+                let working_copy_path = working_copy_path.to_str().ok_or_else(|| {
+                    eyre::eyre!(
+                        "Path to Git repo could not be converted to UTF-8 string: {:?}",
+                        working_copy_path
+                    )
+                })?;
+                result.extend(vec!["--work-tree", working_copy_path]);
```

Comment:
I needed to add this, else `git status` fails with "fatal: This operation must be run in a work tree"

Reply:
Is this still true? I commented this out and the amend tests still passed.

Reply:
I see. `repo_path` here is passed a directory like `foo/bar/.git`, which is not legal to run `git status` inside of. I think the right thing to do is to just set `-C` to `repo.get_working_copy_path()` if it's present, or `repo_path` if not.

Reply:
I suppose it's also a mistake that the process invocation can return error without it being detected. The caller should have to opt into ignoring the exit code (which is the right thing to do in certain cases, such as when running user-defined hooks, but not always).

Reply:
> I think the right thing to do is to just set -C to repo.get_working_copy_path() if it's present, or repo_path if not

Done. Also filed #200